vm/src/pyobject.rs

```diff
@@ -1562,16 +1558,31 @@ impl PyObject {
     }
 
     pub fn payload<T: PyValue>(&self) -> Option<&T> {
-        self.payload.downcast_ref()
+        let payload: &dyn Any = &self.payload;
+        payload.downcast_ref()
     }
 }
 
-// The intention is for this to replace `PyObjectPayload` once everything is
-// converted to use `PyObjectPayload::AnyRustvalue`.
-pub trait PyValue: Any + fmt::Debug {
+pub trait PyValue: fmt::Debug + 'static {
     fn required_type(ctx: &PyContext) -> PyObjectRef;
 }
 
+pub trait PyValuePayload: Any + fmt::Debug + 'static {
+    fn required_type(&self, ctx: &PyContext) -> PyObjectRef;
+}
+
+impl<T: PyValue + 'static> PyValuePayload for T {
+    fn required_type(&self, ctx: &PyContext) -> PyObjectRef {
+        T::required_type(ctx)
+    }
+}
+
+impl PyValuePayload for () {
```

Comment:
What's this impl needed for?

Reply:
Y'know, I could have sworn there was some function that made a PyObject with the payload as `()`, but I guess not.